### PR TITLE
Ft auth test suites

### PR DESCRIPTION
**Pull request by darothub**
### What this PR does (required):
- Test for /register. /user, /login, and /logout end points

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action

### Any information needed to test this feature (required):
- `npm run test` 
- See `README.md` for more details

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ if (process.env.NODE_ENV === "production") {
   );
 } else {
   app.get("/", (req, res) => {
+    console.log(app.get("env"));
```
Comment (moffatethan) [inline server/app.js:55]: This could be removed

File: server/test/auth/auth.spec.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    password: "password",
+  });
+  const token = generateToken(user._id);
+  describe("/POST /auth/register", () => {
+    it("it should return newly created user", (done) => {
+      const user = {
+        username: "teste",
+        email: "test12@email.com",
+        password: "passwordlongone1",
+      };
```
Comment (moffatethan) [inline server/test/auth/auth.spec.js:31]: We could move this in the main scope and re-use it for all the test cases below instead of repeating it

**Review by moffatethan — COMMENTED** [review]
Overall good work, I left a comment there for you to address

File: server/app.js (hunk 1 above)
Comment (darothub) [reply server/app.js:55]: done

File: server/test/auth/auth.spec.js (hunk 2 above)
Comment (darothub) [reply server/test/auth/auth.spec.js:31]: Actually, if you observe the object carefully you will notice that they are not the same. I created them for different scenarios. After the first one,
```
      const user = {
        username: "teste",
        email: "test12@email.com",
        password: "passwordlongone1",
      };
```
 others were changed to test for scenarios as  listed below.
`"it should return status code 400 for existing email"
     const user = {
        username: "testes",
        email: ~~"test12@email.com"~~,
        password: "passwordlongone1",
      };
 `
`"it should return status code 400 for existing username"`
   const user = {
          username: ~~"teste"~~,
          email: "test121@email.com",
          password: "passwordlongone1",
        };
 `

**Review by moffatethan — APPROVED** [review]

**Pull request merged**